src/vulkan-renderer/renderer.cpp

```diff
@@ -109,25 +109,11 @@ VkResult VulkanRenderer::create_command_buffers() {
     spdlog::debug("Allocating command buffers.");
     spdlog::debug("Number of images in swapchain: {}.", swapchain->get_image_count());
 
-    VkCommandBufferAllocateInfo command_buffer_allocate_info = {};
-
-    command_buffer_allocate_info.sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_ALLOCATE_INFO;
-    command_buffer_allocate_info.pNext = nullptr;
-    command_buffer_allocate_info.commandPool = command_pool->get();
-    command_buffer_allocate_info.level = VK_COMMAND_BUFFER_LEVEL_PRIMARY;
-    command_buffer_allocate_info.commandBufferCount = swapchain->get_image_count();
-
-    command_buffers.clear();
+    // TODO: Do we need to clear this when recreating swapchain before filling again?
     command_buffers.resize(swapchain->get_image_count());
 
-    VkResult result = vkAllocateCommandBuffers(vkdevice->get_device(), &command_buffer_allocate_info, command_buffers.data());
-    vulkan_error_check(result);
-
     for (std::size_t i = 0; i < command_buffers.size(); i++) {
-        std::string command_buffer_name = "Command buffer " + std::to_string(i) + " for core engine.";
-
-        debug_marker_manager->set_object_name(vkdevice->get_device(), (std::uint64_t)(command_buffers[i]), VK_DEBUG_REPORT_OBJECT_TYPE_COMMAND_BUFFER_EXT,
-                                              command_buffer_name.c_str());
+        command_buffers[i] = std::make_unique<wrapper::CommandBuffer>(vkdevice->get_device(), command_pool->get());
```

Comment:
I think this would look nicer being:
```c++
command_buffers.emplace_back(new wrapper::CommandBuffer(vkdevice->get_device(), command_pool->get()));
```
although if you remove the `unique_ptr`, it could be:
```c++
command_buffers.emplace_back(vkdevice->get_device(), command_pool->get());
```

Reply:
Done.